src/repository/conUOpenDataCourseRepository.js

```diff
@@ -0,0 +1,120 @@
+const {createOpenDataCourse} = require("../factory/conUOpenDataCourseFactory");
+const OpenDataCourse = require('../models/conUOpenDataCourse.model');
+const axios = require("axios");
+
+module.exports = class OpenDataCourseRepository {
+    /**
+     * Create a course record.
+     * @param {*} data The body/params of the request. It should contain: ID, title, subject, catalog, career, classUnit, prerequisite, crosslisted.
+     * @returns {Course}, Returns a promise. Resolves with the Course.
+     */
+    static create(data) {
+        return new Promise((resolve, reject) => {
+            const newOpenDataCourse = createOpenDataCourse(data)
+            newOpenDataCourse.save((err, faculty) => {
+                if (err) {
+                    reject(err);
+                }
+                resolve(faculty);
+            })
+        })
+    }
+
+    /**
+     * Find all courses with a specific course code.
+     * @param {string} courseCode, the course code of interest.
+     * @returns [{Course}] Returns a promise. Resolves with an array of Courses associated with the courseCode param.
+     */
+    static findAllByCourseCode(courseCode) {
+        return new Promise((resolve, reject) => {
+            OpenDataCourse.find({
+                subject: {"$in": [courseCode.toUpperCase()]}
+            })
+                .then((result) => {
+                    resolve(result);
+                })
+                .catch(err => reject(err))
+        })
+    }
+
+    /**
+     * Find a specific course with a specific course code and course number.
+     * @param {string} courseCode the course code of interest.
+     * @param {string, int} courseNumber the course number of interest.
+     * @returns [{Course}] Returns a promise. Resolves with an array of Course associated with the courseCode and courseNumber param.
```

Comment:
{[Course]}

Reply:
The reason I wrote it like this is because IntelliJ doesn't like it:
![image](https://user-images.githubusercontent.com/72574688/204106581-505857d6-d591-4961-b099-6f4cad45bc27.png)

Will modify, though.